microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java

```diff
@@ -0,0 +1,121 @@
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsBasicEndpoint;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.log.ModelNodeLogChecker;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+/**
+ * Set of tests verifying functionality of {@code mp.jwt.verify.publickey} property.
+ */
+@RunWith(Arquillian.class)
+public class PublicKeyPropertyTestCase {
+
+    private static final String DEPLOYMENT_WITH_VALID_KEY = "valid-key-deployment";
+    private static final String DEPLOYMENT_WITH_INVALID_KEY = "invalid-key-deployment";
+
+    private static RsaKeyTool keyTool;
+
+    @BeforeClass
+    public static void beforeClass() throws URISyntaxException {
+        final URL privateKeyUrl = BasicCdiTest.class.getClassLoader().getResource("pki/key.private.pkcs8.pem");
+        if (privateKeyUrl == null) {
+            throw new IllegalStateException("Private key wasn't found in resources!");
+        }
+        keyTool = RsaKeyTool.newKeyTool(privateKeyUrl.toURI());
+    }
+
+    /**
+     * A deployment with valid key. This means, that it is correct key which can be used to verify the JWT signature.
+     * @return a deployment
+     */
+    @Deployment(name = DEPLOYMENT_WITH_VALID_KEY)
+    public static WebArchive createValidKeyDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClass(JaxRsBasicEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(BasicCdiTest.class.getClassLoader().getResource("mp-config-pk-valid.properties"),
+                        "microprofile-config.properties");
+    }
+
+    /**
+     * A deployment with invalid key. The verification fails when this key is used.
+     * @return a deployment
+     */
+    @Deployment(name = DEPLOYMENT_WITH_INVALID_KEY)
+    public static WebArchive createInvalidKeyDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClass(JaxRsBasicEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(BasicCdiTest.class.getClassLoader().getResource("mp-config-pk-invalid.properties"),
+                        "microprofile-config.properties");
+    }
+
+    /**
+     * @tpTestDetails A request with proper JWT is sent on server which has configured bad public key. The server fails
```

Comment:
Add negative scenario, please.

Reply:
Thanks for suggestion! Can you please describe such scenario?